### ISSUE TITLE
Better support for multi-package repositories, e.g. Lerna

### ISSUE THREAD

**Issue by jxpx777** [vim-test/vim-test#489]
I'm working on a project where we're using Lerna to manage a repository of several different packages (client, GraphQL layer, data gateway, etc). I like to work from my repository root since that's where my ctags get generated and tucked away inside the `.git` directory. When I do, my Jest tests (`foo.test.js`) don't get recognized as tests because the root package.json does not include this as a dependency; the GraphQL layer does. If I `:chdir packages/api`, my test file is recognized properly.

Option 1: Look at the Lerna config and reason about packages from there. (Seems kind of icky tbh…)
Option 2: Work from the dirname of the current file _upward_ to find the first package.json and check _that one_ rather than looking for a package.json in the current working directory.

**Comment by codeinabox** [comment]
This is quite similar to #272. I agree that option one is icky and too much complexity. There is a pull request #410 to implement option two so I'll see if that's something we can consider.

I have found in general that many vim plug-ins don't play so well with mono repositories as often they assume the current working directory is the package root. I've been able to get around this by opening each package in a different tab and using `tcd` in Neovim to change the working directory just for that tab.
